src/server/services/saml.strategy.ts

```diff
@@ -0,0 +1,47 @@
+import { Strategy } from 'passport-saml';
+import { PassportStrategy } from '@nestjs/passport';
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { UserData, HarvardKeyProfile } from '../interfaces';
+import { ConfigService } from './config.service';
+
+/**
+ * Implements passport-saml to connect to Harvard Key for authentication
+ */
+
+@Injectable()
+class SAMLStrategy extends PassportStrategy(Strategy) {
+  private readonly devMode: boolean;
+
+  public constructor(config: ConfigService) {
+    super({
+      entryPoint: config.get('CAS_URL'),
+      issuer: 'passport-saml',
+      host: config.get('EXTERNAL_URL'),
+    });
+    this.devMode = !config.isProduction;
+  }
+
+  public async validate(profile: HarvardKeyProfile): Promise<UserData> {
+    if (this.devMode) {
+      return {
+        id: 'abc123',
+        firstName: 'Test',
+        lastName: 'User',
+        email: 'noreply@seas.harvard.edu',
+        displayName: 'testUser',
+      };
+    }
+    if (!profile) {
+      throw new UnauthorizedException();
```

Comment:
I would provide some kind of message here that we can display on the client side that provides more context...

```suggestion
      throw new ForbiddenException('You are not authroized to use this application');
```